include/unifex/any_scheduler.hpp

```diff
@@ -0,0 +1,185 @@
+
+template <typename... CPOs>
+using _void_receiver_ref = _any::_receiver_ref<type_list<CPOs...>>;
+
+template <typename... CPOs>
+struct _schedule_and_connect_fn {
+  struct type {
+    using _rec_ref_t = _void_receiver_ref<CPOs...>;
+    using type_erased_signature_t = _any::_operation_state(this_ const&, _rec_ref_t);
+
+    template (typename Scheduler)
+      (requires tag_invocable<type, Scheduler const&, _rec_ref_t>)
+    _any::_operation_state operator()(Scheduler const& sched, _rec_ref_t rec) const {
+      return tag_invoke(*this, sched, (_rec_ref_t&&) rec);
+    }
+
+    template (typename Scheduler)
+      (requires (!tag_invocable<type, Scheduler const&, _rec_ref_t>) AND
+        scheduler<Scheduler>)
+    _any::_operation_state operator()(Scheduler const& sched, _rec_ref_t rec) const {
+      return _any::_connect<type_list<CPOs...>>(schedule(sched), (_rec_ref_t&&) rec);
+    }
+  };
+};
+
+template <typename... CPOs>
+inline constexpr typename _schedule_and_connect_fn<CPOs...>::type _schedule_and_connect{};
+
+template <typename... CPOs>
+using _any_void_sender_of =
+  typename _any::_with_receiver_queries<CPOs...>::template any_sender_of<>;
+
+template <typename... CPOs>
+using any_scheduler_impl =
+  any_unique_t<
+    _schedule_and_connect<CPOs...>,
+    _copy_as<any_scheduler<CPOs...>>,
+    _get_type_id,
+    overload<bool(const this_&, const any_scheduler<CPOs...>&)>(equal_to)>;
+
+template <typename... CPOs>
+struct _any_scheduler<CPOs...>::type {
+  template (typename Scheduler)
+    (requires (!same_as<Scheduler, type>) AND scheduler<Scheduler>)
+  /* implicit */ type(Scheduler sched)
+    : impl_((Scheduler&&) sched) {}
+
+  type(type&&) noexcept = default;
+  type(const type& that)
+    : impl_(_copy_as<type>(that.impl_).impl_) {}
+
+  type& operator=(type&&) noexcept = default;
+  type& operator=(const type& that) {
+    impl_ = _copy_as<type>(that.impl_).impl_;
+    return *this;
+  }
+
+  struct _sender {
+    template <template <class...> class Variant, template <class...> class Tuple>
+    using value_types = Variant<Tuple<>>;
+
+    template <template <class...> class Variant>
+    using error_types = Variant<std::exception_ptr>;
+
+    static constexpr bool sends_done = true;
+
+    _sender(_sender&&) noexcept = default;
+
+    template (typename Receiver)
+      (requires receiver_of<Receiver> AND
+        (invocable<CPOs, Receiver const&> &&...))
+    any_operation_state_for<Receiver> connect(Receiver rec) && {
+      any_scheduler_impl<CPOs...> const& impl = impl_;
+      return any_operation_state_for<Receiver>{
+          (Receiver&&) rec,
+          [&impl](_void_receiver_ref<CPOs...> rec2) {
+            return _schedule_and_connect<CPOs...>(
+                impl, (_void_receiver_ref<CPOs...>&&) rec2);
+          }
+        };
+    }
+
+  private:
+    friend type;
+    _sender(any_scheduler_impl<CPOs...> const& impl) noexcept
+      : impl_(impl)
+    {}
+    any_scheduler_impl<CPOs...> const& impl_;
```

Comment:
This is technically a lifetime issue. There is no requirement that the scheduler remain alive until the result of `schedule` is connected. This needs a bit of thought. Ideally, we would be able to either:
* hold a strong reference to the impl when it is dynamically-allocated, or
* store a copy of the impl when it is stored in situ (one we implement that).